presto-hive-common/src/main/java/com/facebook/presto/hive/aws/security/lakeformation/AWSLakeFormationSecurityMapping.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.aws.security.lakeformation;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class AWSLakeFormationSecurityMapping
```

Comment:
It looks like AWSLakeFormationSecurityMapping and AWSS3SecurityMapping should have a common, possibly abstract, SecurityMapping superclass. That might allow you to avoid some of the problems in AWSSecurityMappings which could work exclusively with the supertype.

Reply:
Sure, let me see what is the best way to simplify the implementation.